### DISCUSSION TITLE
Selector requirements for Divina

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/annotations#5] [General]
Under the current draft, we have the following requirements for Divina:

> A Divina annotation:
>
>- MUST contain a Rectangular Media FragmentSelector,
>- MAY contain a ProgressionSelector.

I think that this might limit our ability to annotate content in a Divina publication. A user might want to:

- annotate an entire image (especially for paged publications where each page is an image)
- an entire spread
- or a fragment over an entire spread

This also makes me wonder about the implementation of Divina annotations using a rectangular media fragment:

- is this meant to align with Guided Navigation Documents? If so, shouldn't we also support polygonal/multi-points as well?
- or does this mean that the user would have to draw a rectangle first before adding an annotation?

If this is meant to align with Guided Navigation Documents, there might be other ways of doing so. If this requires drawing a rectangle, I'd be worried about the UX on mobile devices.

**Comment by llemeurfr** [comment]
In the [W3C Annotation Model](https://www.w3.org/TR/annotation-model/#selectors), I read "Many Annotations refer to part of a resource, rather than all of it, as the Target.." Not ALL, but Many. 
In this model nothing forbids having a `target` with a `source` but no `selector`. 

We just have to make Selector not required in https://github.com/readium/annotations?tab=readme-ov-file#12-target and add some wording about the behavior of a target with no selector. 
Would it be ok?

**Comment by llemeurfr** [comment]
@HadrienGardeur , about your comment on annotating spreads (= 2 images at the same time) and parts of a spread (an area spreading between two images) : 
In https://www.w3.org/TR/annotation-model/#cardinality-of-bodies-and-targets, it is specified that an annotation can have multiple targets.
Therefore, this flexibility is possible, but as usual, it comes with a price: more complexity.
I'm not convinced that having this in a v1 is a good move. 

Re. polygonal regions, I don't find any spec other than https://idpf.org/epub/renditions/region-nav/#sec-3.5.1.3 which has never been implemented, to my knowledge.
